src/fn/upload-partners.js

```diff
@@ -0,0 +1,42 @@
+const attachmentsFromDir = require('../lib/attachments-from-dir');
+const environment = require('../lib/environment');
+const fs = require('../lib/sync-fs');
+const pouch = require('../lib/db');
+const insertOrReplace = require('../lib/insert-or-replace');
+const warnUploadOverwrite = require('../lib/warn-upload-overwrite');
+const { info, warn } = require('../lib/log');
+
+module.exports = {
+  requiresInstance: true,
+  execute: async () => {
+    const partnersPath = fs.path.resolve(`${environment.pathToProject}/partners.json`);
+
+    if (!fs.exists(partnersPath)) {
+      warn(`No partners file found at path: ${partnersPath}`);
+      return Promise.resolve();
+    }
+
+    const partnersSettings = fs.readJson(partnersPath);
+
+    const doc = {
+      _id: 'partners',
+      resources: partnersSettings.resources,
+      _attachments: attachmentsFromDir(`${environment.pathToProject}/partners`),
+    };
+
+    const db = pouch();
+
+    const changes = await warnUploadOverwrite.preUploadDoc(db, doc);
+
+    if (changes) {
+      await insertOrReplace(db, doc);
+      info('Partners file uploaded');
+    } else {
+      info('Partners file not uploaded as no changes found');
+    }
+
+    warnUploadOverwrite.postUploadDoc(doc);
+
+    return Promise.resolve();
+  }
+};
```

Comment:
This is so similar to `upload-branding` and `upload-resources` there should be some way to make a lib function which all three can call to encourage reuse. At least lines 27-40 can be extracted, but I think if you may be able to pull out a function which takes the JSON path, the attachments path, and the base doc as parameters so this file just calls that function and then logs the outcome.

Reply:
Done!

Reply:
Awesome!